torch/_ops.py

```diff
@@ -32,14 +34,21 @@ class OpOverload:
     def __init__(self, overloadpacket, op, schema, tags):
         self._op = op
         self._schema = schema
+
+        self._parsed_schema: FunctionSchema = None
+        try:
+            self._parsed_schema: FunctionSchema = FunctionSchema.parse(str(schema))
+        except Exception as e:
+            # TODO: fix here
```

Comment:
Running into issue here https://github.com/pytorch/pytorch/issues/83214

Reply:
What do you think about parsing the schema for OpOverload? 
Shall I go ahead and fix the parsing issue?

Reply:
@ezyang 

we have two version of FunctionSchema: torch._C.FunctionSchema and torchgen.model.FunctionSchema

and IIUC, we would prefer to use the second one more widely, as it's in python, and it's more flexible

However, OpOverload is currently using the first one, and it's pretty widely referenced across the code base...

Reply:
Parsing into torchgen is fine. The history here is that the C function schema first, and then the torchgen schema came later but it was not importable from torch proper, and now it only became importable recently.

Reply:
btw, this is what executorch has been doing to get schema access, @shunting314 and @larryliu0820 can tell you more about it ;)

Reply:
> btw, this is what executorch has been doing to get schema access

yes, we are doing the same thing in executorch. We also convert an _C.FunctionSchema to torchgen.FunctionSchema so we can use torchgen API to check if an op is out variant and convert it to out variant otherwise.

I've seen most of the issues listed here https://github.com/pytorch/pytorch/issues/83214 :) E.g., here are the categories of issues I seen:
- FunctionSchema parser does not recognize types (like complex etc.)
- FunctionSchema argument parser does not support type/default value containing comma (E.g. `Dict(str, int)`, `[1, 2]` ). The reason is the argument parser split the argument string by ',' to get argument list. The comma in a type or default value causes issue here. This same issue cause all the 'ValueError: not enough values to unpack (expected 2, got 1)'  error in the list.
- There is one more issue for Tensor list. The alias annotation will get lost when serializing a _C.FunctionSchema containing a tensor list with alias annotation on tensor. Tracked here: https://github.com/pytorch/pytorch/issues/77920 

In our case, the parse failures for the first 2 categories are simply ignored since those ops are not really used in the model. For the last category, since there are not many ops like that, we right now have a hack list to manually patch the schema string from _C.FunctionSchema serialization.

Reply:
> @shunting314 I think it might be worth to properly solve these issues, align _C.FunctionSchema with torchgen.model.FunctionSchema. I guess torchgen should be able to catch up with _C.FunctionSchema features.

I'll certainly be happy to see the full parity between _C.FunctionSchema and torchgen FunctionSchema. Actually Brian and I have already fixed a few issues previously. The rest ones above should also be solvable if people have bandwidths to fix them now.

One thing to call out about the 2nd category,  I think using an LL parser can handle the special cases mentioned. But it will be slower than simply splitting by comma.  If we don't want to slow down codegen, maybe we can keep both:
- use LL parser to parse serialized _C.FunctionSchema
- splitting by comma for schema strings from yaml file.